src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,68 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserAgeException;
+import core.basesyntax.exception.UserLoginExistsException;
+import core.basesyntax.exception.UserLoginLengthException;
+import core.basesyntax.exception.UserPasswordLengthException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_LOGIN_LENGTH_MIN = 6;
+    private static final int USER_PASSWORD_LENGTH_MIN = 6;
+    private static final int USER_AGE_MIN = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        String userLogin = user.getLogin();
+        String userPassword = user.getPassword();
+        int userAge = user.getAge();
+        if (isUserLoginExist(userLogin)) {
+            throw new UserLoginExistsException("User with this login: "
+                                                       + userLogin
+                                                       + " already exists");
```

Comment:
The use of 'else if' after a throw statement is unnecessary and can be replaced with a simple 'if'. Each check should be a separate 'if' statement.